python/singa/autograd.py

```diff
@@ -1828,3 +1828,18 @@ def backward(self, dy):
 
 def leakyrelu(x, a=0.01):
     return LeakyRelu(a)(x)[0]
+
+
+class Sign(Operation):
+    def forward(self, a):
+        if training:
+            self.input = a
+        return singa.Sign(a)
+
+    def backward(self, dy):
+        dx = singa.MultFloat(dy, 0.0)
+        return dx
+
+
+def sign(a):
+    return Sign()(a)[0]
```

Comment:
change a to x
def sign(x):
    return Sign()(x)[0]

Reply:
input of the function is "a"